### ISSUE TITLE
Moving SDS to Beta

### ISSUE THREAD

**Issue by costinm** [istio/istio#15267]
Epic for grouping all required and optional steps for beta/production use of SDS.

This is specific for use of SDS for Sidecars -  use of SDS for Ingress secrets is already relatively 
stable and bugs/gaps for SDS-for-ingress should be addressed in separate issues.

[ ] SDS must be used for communication with the XDS server (pilot)
[ ] SDS enabled in Pilot, Mixer, Galley
[ ] Gradual adoption of SDS, by upgrading a cluster without SDS and migrating to SDS
[ ] Metrics, readiness for SDS components
[ ] Tools and documentation for troubleshooting

**Comment by incfly** [comment]
Adding one more issue, (sorry not sure how to mark it as epic blocker)
https://github.com/istio/istio/issues/15701#issuecomment-516794648

TLDR is to ensure CNI working well with SDS.

**Comment by howardjohn** [comment]
This is done
